src/target/jtag_scan.c

```diff
@@ -210,9 +210,17 @@ static void jtag_display_idcodes(void)
 
 static jtag_ir_quirks_s jtag_device_get_quirks(const uint32_t idcode)
 {
+	static size_t previous_idx = 0;
+
 	for (size_t idx = 0; dev_descr[idx].idcode; ++idx) {
 		if ((idcode & dev_descr[idx].idmask) == dev_descr[idx].idcode)
+		{
+			/* workaround for ESP32-P4 which has 2 different ir_value */
+			if ((idcode == 0x12c25) && (previous_idx == idx))
```

Comment:
Please drop the extra parens - they are not only not required, they actively suppress a warning about accidental assignment. Please suffix the integer constant with `U` to make it unsigned.